expyfun/_experiment_controller.py

```diff
@@ -1101,6 +1106,54 @@ def check_force_quit(self):
         """Check to see if any force quit keys were pressed."""
         self._response_handler.check_force_quit()
 
+    def text_input(self, stop_key='return', pos=[0, 0], color='white',
+                   font_name='Arial', font_size=24, wrap=True, units='norm',
+                   all_caps=True):
+        """Allows participant to input text and view on the screen.
+
+        Parameters
+        ----------
+        stop_key : str
+            The key to exit text input mode.
+        pos : list | tuple
+            x, y position of the text. In the default units (-1 to 1, with
+            positive going up and right) the default is dead center (0, 0).
+        color : matplotlib color
+            The text color.
+        font_name : str
+            The name of the font to use.
+        font_size : float
+            The font size (in points) to use.
+        wrap : bool
+            Whether or not the text will wrap to fit in screen, appropriate
+            for multi-line text. Inappropriate for text requiring
+            precise positioning or centering.
+        units : str
+            Units for `pos`. See `check_units` for options. Applies to
+            `pos` but not `font_size`.
+        all_caps : bool
+            Whether the text should be displayed in all caps.
+        """
+        letters = string.ascii_letters
+        text = str('')
+        while True:
+            letter = self.wait_one_press(timestamp=False)
+            if letter == stop_key:
+                self.flip()
+                break
+            if letter == 'backspace':
+                text = text[:-1]
+            else:
+                letter = ' ' if letter == 'space' else letter
+                letter = letter.upper() if all_caps else letter
+                text += letter if letter in letters or letter == ' ' else ''
```

Comment:
optional: this might be a little clearer if, up above:
`letters = string.ascii_letters + ' '`
and then this becomes
`text += letter if letter in letters else ''`

Reply:
makes sense